src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ColorFilter.tsx

```diff
@@ -121,16 +122,19 @@ export const ColorFilter: React.FC<ColorFilterProps> = ({
       COLOR_OPTIONS.slice(INITIAL_ITEMS_TO_SHOW).map(({ value }) => value)
     ).length > 0
   const hasColorFilter = currentlySelectedFilters().colors.length > 0
-
+  const resultsSorted = sortResults(
+    currentlySelectedFilters().colors,
+    COLOR_OPTIONS
+  )
   return (
     <Expandable mb={1} label="Color" expanded={hasColorFilter || expanded}>
       <Flex flexDirection="column">
         <ShowMore expanded={hasBelowTheFoldColorFilter}>
-          {COLOR_OPTIONS.map(colorOption => {
+          {resultsSorted.map(colorOption => {
             return (
               <ColorFilterOption
                 key={colorOption.value}
-                colorOption={colorOption}
+                colorOption={colorOption as any}
```

Comment:
so, the resulting type of `sortResults` is `Result`, which doesn't include `hex` (the error here). But in reality the return type of `sortResults` should match whatever `COLOR_OPTIONS` is (the passed-in second argument). I spent way too long trying to futz with the typings to get that working...I have much to learn yet 😅

Reply:
Yeah, it'd be something like:

```tsx
const sortResults = <T,>(
  selectedValues: Array<T>,
  allResults: Array<Result<T>>
)
```

```tsx
type Result<T> = {
  name: string
  count: number
  value: T
}
```

if I'm understanding correctly